Source/Classes/PINRemoteImageManager.m

```diff
@@ -784,13 +786,62 @@ - (void)downloadImageWithURL:(NSURL *)url
             NSError *remoteImageError = error;
             PINImage *image = nil;
             id alternativeRepresentation = nil;
-             
+            __block NSNumber *maxAge = nil;
             if (remoteImageError == nil) {
-                 //stores the object in the caches
-                 [self materializeAndCacheObject:data cacheInDisk:data additionalCost:0 url:url key:key options:options outImage:&image outAltRep:&alternativeRepresentation];
+                if (_isTtlCache && !(options & PINRemoteImageManagerDownloadOptionsIgnoreCacheControlHeaders)) {
+                    // examine Cache-Control headers (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)
+                    if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+                        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
+
+                        [[[httpResponse allHeaderFields][@"Cache-Control"] componentsSeparatedByString:@","] enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+                            NSString *trimmed = [[(NSString *) obj stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]] lowercaseString];
+
+                            if ([trimmed isEqualToString:@"no-store"] || [trimmed isEqualToString:@"must-revalidate"] || [trimmed isEqualToString:@"no-cache"]) {
+                                maxAge = @(0);
+                                *stop = YES;
+                            } else {
+                                // max-age
+                                NSArray<NSString *> *split = [trimmed componentsSeparatedByString:@"max-age="];
+                                if ([split count] == 2) {
+                                    maxAge = @([split[1] integerValue]);
+                                }
+                            }
+                        }];
+
+                        NSString *expires = nil;
+                        // If there is a Cache-Control header with the "max-age" directive in the response, the Expires header is ignored
+                        if (!maxAge && (expires = [httpResponse allHeaderFields][@"Expires"])) {
```

Comment:
Definitely move this expires = assignment up to the declaration...although it can be useful and crafty in some cases, I think there's no reason to put the assignment in the conditional in this case.

Reply:
I was mimicking the style used in `-[PINRemoteImageDownloadTask didReceiveResponse:]`, although it is far enough away that maybe we don't care (or maybe want to update *that* to use separate assignments)